workflows/offline_ml_diags/offline_ml_diags/create_report.py

```diff
@@ -86,43 +87,25 @@ def _create_arg_parser() -> argparse.Namespace:
     return parser.parse_args()
 
 
-if __name__ == "__main__":
-
-    logger.info("Starting create report routine.")
-    args = _create_arg_parser()
-
-    temp_output_dir = tempfile.TemporaryDirectory()
-    atexit.register(_cleanup_temp_dir, temp_output_dir)
-
-    ds_diags, ds_diurnal, ds_transect, metrics, config = open_diagnostics_outputs(
-        args.input_path,
-        diagnostics_nc_name=NC_FILE_DIAGS,
-        diurnal_nc_name=NC_FILE_DIURNAL,
-        transect_nc_name=NC_FILE_TRANSECT,
-        metrics_json_name=JSON_FILE_METRICS,
-        config_name="config.yaml",
-    )
-    ds_diags = ds_diags.pipe(insert_dataset_r2).pipe(mse_to_rmse)
+def render_model_sensitivity(figures_dir, output_dir) -> str:
+    report_sections: MutableMapping[str, Sequence[str]] = {}
 
-    # omit physics tendencies from report plots
-    ds_diags = drop_physics_vars(ds_diags)
-    ds_diurnal = drop_physics_vars(ds_diurnal)
+    for png in copy_pngs_to_report(figures_dir, output_dir):
+        report_sections[png] = [png]
 
-    # diagnostics_utils currently fill dQ1/2 with zeros if not predicted
-    # exclude these from the report if they are not model outputs.
-    ds_diags = drop_temperature_humidity_tendencies_if_not_predicted(
-        ds_diags, config["output_variables"]
-    )
-    ds_diurnal = drop_temperature_humidity_tendencies_if_not_predicted(
-        ds_diurnal, config["output_variables"]
+    return report.create_html(
+        sections=report_sections, title="Model sensitivity to inputs",
     )
 
-    config.pop("mapping_kwargs", None)  # this item clutters the report
-    if args.commit_sha:
-        config["commit"] = args.commit_sha
 
+def render_index(metrics, ds_diags, ds_diurnal, ds_transect, output_dir) -> str:
```

Comment:
This function is the original code from `__main__` refactored out into its own function.